tests/test_components.py

```diff
@@ -84,3 +86,23 @@ def test_real_estate_area():
     RealEstateArea(
         value=123.45, unitCode="SQMT", raw="123.45 square meters", areaType="LOT"
     )
+
+
+def test_headers():
+    Header(name="Connection", value="keep-alive")
+    Header(name="Content-Type", value="text/html; charset=utf-8")
+
+
+def test_navigation():
+    Navigation(url="https://example.com")
```

Comment:
I guess that would depend if we will implement Browser Request on the future and how we do it? 
Will it be part of the current `Navigation` object? Will it be an independent class?

If we use a single class for all then `Request` looks good. Otherwise, we should have some references to http and browser on the class name.

Reply:
I think we can discern them via `HttpRequest` and `BrowserRequest`. Probably something like https://github.com/scrapy/scrapy/blob/master/scrapy/http/response/html.py#L11

Reply:
Let's clarify what we mean by "Browser Request" :) I think there are at least 2 ways to understand it:

1. When you return a browser request e.g. `BrowserRequest(url=...)`, then browser should be used to render the *target* page.
2. There are some additional request parameters, like `actions` or `js_code`, which you need to execute on the *original page* to get to the next page. For example, click on some button.

What of these use cases do you have in mind @VMRuiz @Gallaecio @BurnzZ ?

My 2c: I'm not sure (1) is needed. It seems it should be a decision made separately if we need to use a browser on the *target* page or not, it's not something you should decide during the navigation. Also, it doesn't solve a case when there is no real link available on a page, but there is a way to go e.g. to a next page by performing some JS action.

(2) may require some thought. It should work, but it'd mean that we might be rendering the page 1 more time than necessary, though this can be ok.

Reply:
If we're talking about (2), then both Request and Navigation looks ok to me, with the idea that we can extend them in future, adding a field like `browserActions`.

Reply:
We also have the existing Link data type, which is similar; that's another name we can use.

Reply:
Case 2 is obviously need for: 
- Javascript pagination that is too complex to easily replicate with HTTP
- Filling input forms that load dynamically or contains sliders. For example, delivery food websites with location inputs that display a map for address selection, etc...   

In this case it would means: "Load page A and execute these XYZ actions to get to the next desired page" 

Case 1 is a lit bit more complex because I agree that the Page Objects should not be responsible of knowing if the next page would require rendering to fetch data properly. However, we still need some way to indicate to the crawler that page must be rendered. Maybe, that can be define as a dependency: 

- Page Object A define its WebPage must come from a rendering process
- Scrapy detects a Navigation with a URL + Callback that would trigger Page Object A 
- Scrapy generates a Request with rendering enabled

The alternative would be:

- Always enabled rendering by default  - maybe configurable by netlocs -  
- Crawl non-rendered version by default, and then retry from the Page Object that require rendering

Both options seems very inefficient

Reply:
> However, we still need some way to indicate to the crawler that page must be rendered.

@VMRuiz if a page object needs to use browser to render the page and extract the data, it should use [BrowserHtml](https://web-poet.readthedocs.io/en/stable/api-reference.html#web_poet.page_inputs.browser.BrowserHtml) dependency instead of HttpResponse dependency.

Reply:
Great, but we still a way to tell Scrapy to use browser rendering to build the BrowserHtml object

Reply:
Yes; it's a matter of defining a scrapy-poet provider for BrowserHtml. Probably, we should put the implementation of this provider to scrapy-zyte-api package. It's not implemented yet. See https://scrapy-poet.readthedocs.io/en/stable/providers.html (async def `__call__` should be supported, if I'm not mistaken).

Reply:
So, do we all agree with naming this `Request` with the possibility of adding `actions` field on the future?

Reply:
Component `Navigation` is now `Request`